### DISCUSSION TITLE
Yarn 2 (berry) compatibility

### DISCUSSION THREAD

**Discussion by fjavierv** [mlaursen/react-md#1226] [General]
Hi @mlaursen , I am trying to use react-md with yarn berry. After #1224 now it is possible to install correctly react-md using yarn berry, however when I try to use webpack or tsc I found more problems about depencencies.

I would like to help you to resolve this with some pull request, but before, I have some questions about the current dependencies in react-md.

1. I saw that there are two versions of react and react-dom in some packages, for example in chip package:

```json
{
"dependencies": {
    "@react-md/elevation": "^3.0.1",
    "@react-md/icon": "^3.0.0",
    "@react-md/states": "^3.0.1",
    "@react-md/theme": "^3.0.0",
    "@react-md/typography": "^3.0.0",
    "@react-md/utils": "^3.0.0",
    "classnames": "^2.3.1"
  },
  "devDependencies": {
    "react": "^17.0.2",
    "react-dom": "^17.0.1"
  },
  "peerDependencies": {
    "react": ">= 16.8",
    "react-dom": ">= 16.8.0"
  },
  "optionalDependencies": {
    "prop-types": ">= 15.6"
  }
}

```

What is the reason for that? 

2. In yarn 2 (with strict mode) every package should include all the needed dependencies. react-md is a library using types from react, so i think that @types packages should be moved form the root  package.json to the "subpackages".

3. After fork and clone your project, I link it with my project using a new featured yarn link command to prepare my react-md development, but I found problems because the root package.json has the same name than packages/react-md/package.json. I think that would be good to change one of them.

**Comment by mlaursen** [comment]
Hey @fjavierv , thank you for helping out with this. I used [this repo](https://github.com/mlaursen/mlaursen.com) as an example for Yarn Berry to get `react-md` working but never got it fully updated with pnp because of eslint plugin issues. It's probably too small of an example, so I definitely appreciate the help.

To answer your questions:

1. There's no real reason for different dependencies and they can be safely updated. I normally just let dependabot update them or run `yarn upgrade-interactive --latest` so it might've just been missed.
2. That sounds good to me -- I've been meaning to move dependencies from the root `package.json` for awhile since I only added them when trying out `lerna`
3. I think renaming the root `package.json`'s name to `base-react-md` or another name would be fine

**Reply by fjavierv** [reply]
Hi @mlaursen, I was working in the branch "yarn2" of my fork to migrate react-md v3 to yarn 2. It is not finish, but I think that is a good start point. Could you revised from there before a propose de the pull request? Or you prefer some other methods. Thanks in advance.

**Reply by mlaursen** [reply]
Hey @fjavierv, I tool a look at your branch today and it looks to be fine so far. The only thing I noticed was that a few packages now have `react-router-dom` as a `peerDependency` which I don't think should not be required. Maybe it should have been set as a `devDependency` instead?

**Reply by fjavierv** [reply]
Hi @mlaursen, you were right. I remove all `react-router-dom` peerDependencies, now only two package need `react-router-dom` as devDependencies (because unit test). I need to follow testing react-md with yarn2, many development tools have problems with yarn2. It is importan to plan well the migration, maybe we can try to upload changes relates with dependencies and follow testing until jump to yarn2. What do you think about?

**Reply by mlaursen** [reply]
Hey @fjavierv sorry for the delayed response -- unfortunately I think it might be best to put the remaining work on hold for now just to be safe. I found out [dependabot doesn't support yarn 2 yet](https://github.com/dependabot/dependabot-core/issues/1297) which is a blocker for me an the changes in yarn 2 aren't really noticeable/good enough for me to find replacements for all the development tools yet.

**Reply by fjavierv** [reply]
Hi  @mlaursen, sorry for the delay, I have been a father recently. I agree with you, it is better not to use yarn 2 for now. But maybe you can consider to use all the "package.json" changes in this  [commit](https://github.com/fjavierv/react-md/commit/2cb9559e49727efa21190381bdc7d5aedacd8f1d). With this probably we can close this [issue](https://github.com/mlaursen/react-md/issues/1224) and also will be easy to migrate react-md to yarn2 in the future.
